boot/freeldr/freeldr/arch/i386/entry.S

```diff
@@ -47,6 +48,11 @@ _RealEntryPoint:
     mov gs, ax
     mov ss, ax
 
+    /* Keep only the low 16 bits of stack */
+    and esp, HEX(FFFF)
+
+    call FinishRelocation
```

Comment:
Regarding the esp question I previously had, why wouldn't it be possible to setup the pmode stack (what's done below) before this function call?

Reply:
because the protected/long mode will overwrite the Freeldr if it's loaded in `0x7C00`!

Reply:
but this PR doesn't add support for loading in `0x7C00` it's only a preparation for the next PR.